concordium-consensus/src/Concordium/GlobalState/Basic/BlockState/Updates.hs

```diff
@@ -324,14 +343,40 @@ processProtocolUpdates t pu0 uq = (pu', uq {_uqQueue = qr}, Map.fromAscList ql)
         pu' = getFirst <$> mconcat ((First <$> pu0) : (Just . First . snd <$> ql))
 
 
+-- |Process AR and IP update queues.
+-- Ensuring that new IPs/ARs have unique ids is difficult when enqueueing.
+-- Instead, it is handled here by ignoring updates with duplicate IPs/ARs.
+processARsAndIPsUpdates :: Ord k
+                        => Map.Map k v -- ^ The existing IPs / ARs.
+                        -> (v -> k) -- ^ Getter for the key field.
+                        -> Timestamp
+                        -> UpdateQueue v
+                        -> (Map.Map k v, UpdateQueue v, Map.Map TransactionTime v)
+processARsAndIPsUpdates oldValMap getKey t uq = (updatedValMap, uq {_uqQueue = qr}, changes)
+  where (ql, qr) = span ((<= t) . transactionTimeToTimestamp . fst) (uq ^. uqQueue)
+        (changes, updatedValMap) = foldr go (Map.empty, oldValMap) ql
```

Comment:
The use of foldr implicitly makes it so that updates with the later timestamp will be preferred if there is a clash of ids.

Is that the intention? If so, please document the intent.